Automattic/newspack-theme
# Header: Add mobile CTA button

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a mobile-only CTA button to the header.

When activated, it uses a red background by default, and the 'Menu' text next to the mobile menu is hidden.

The button will only display when enabled, and when a page is set for it. 

Some questions for testing:
* Does the CTA option's location in the Customizer make sense? (It's under Header Settings).
* Right now the option forces you to pick a page on the site; will this work, or should it be a text field for a URL instead so it can point to other sites. 
* Does the colour picker location make sense? The header background colour picker lives under the Colors panel, though that's largely due to the fact it's tied to tightly to the other colour settings.

Closes #977.

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Switch to the default Newspack theme.
3. Navigate to Customize > Header Settings > Mobile Call-to-Action.
4. Check 'Show Mobile CTA', and pick a page under 'Button Link' -- the button won't show unless it is linked to something.
5. Navigate to Customize > Header Settings > Appearance, and uncheck 'Short Header' if it's checked.
6. Click 'Publish'.
7. Shrink down the browser window until the mobile menu appears; the Donate link should be next to it:

![image](https://user-images.githubusercontent.com/177561/86168175-a99db080-bacc-11ea-9f95-5c3b9ad9362b.png)

8. Navigate back to Customize > Header Settings > Mobile Call-to-Action and try changing the button text and background color -- if you pick a dark colour, the text should stay white, but if you pick a lighter colour, it should switch to black to maintain contrast.
9. Navigate to Customize > Typography and change the header font; this font should be picked up by the button, along with your text/colour changes:

![image](https://user-images.githubusercontent.com/177561/86168626-55470080-bacd-11ea-98f8-a81dd62df427.png)

10. Navigate to Customize > Header Settings > Appearance, and check 'Short Header' -- the mobile menu appears at a wider breakpoint with this settings, so it's important to make sure the button still appears only when the mobile menu is enabled at that wider breakpoint:

![image](https://user-images.githubusercontent.com/177561/86168797-8b848000-bacd-11ea-9d6e-5e40edf8450e.png)

11. Cycle through the different child themes and confirm that the button displays well; in all cases but the default, it should use squared corners:

**Joseph:**

![image](https://user-images.githubusercontent.com/177561/86169077-e918cc80-bacd-11ea-8301-6c8d23516726.png)

**Katharine:**

![image](https://user-images.githubusercontent.com/177561/86169125-fcc43300-bacd-11ea-953c-7f3a3a9aea60.png)

**Nelson:**

![image](https://user-images.githubusercontent.com/177561/86169166-11083000-bace-11ea-809e-2489c6443d22.png)

**Sacha:**

![image](https://user-images.githubusercontent.com/177561/86169226-25e4c380-bace-11ea-94b5-f99b7221a5ef.png)

**Scott:**

![image](https://user-images.githubusercontent.com/177561/86169305-457bec00-bace-11ea-8433-9824818de9ab.png)

### Other information:
